complete tds #10 and #11, clear td#9

ezisberta opened the pull request:
About the previous feedback: I've added a feature that only let's a user have the button to add a comment if it exists or if it's valid, I took that example from most of the apps I use.

About pulling the main branch, I'm going to do an NC help because I'm afraid of accidentally losing the work I've done locally .

ezisberta committed:
complete td # 9

ezisberta committed:
complete tds #10 and #11, clear td#9

Poonam-raj commented on src/Routed/SingleReviewPage.jsx:
```diff
@@ -51,10 +90,16 @@ export default function SingleReviewPage() {
+              </button>
+            )}
+          </span>
         </div>
```
I would consider extracting the VoteField bit of HTML into its own component to make this component less busy

Poonam-raj commented on src/Routed/SingleReviewPage.jsx:
```diff
@@ -51,10 +90,16 @@ export default function SingleReviewPage() {
+          </span>
         </div>
         <div className="commentSection">
           <h2 className="CommentHeaderWithCount">
```
I would consider extracing the whole 'commentSection' out into its own component too to neaten this component up

Poonam-raj commented on src/Routed/SingleReviewPage.jsx:
```diff
@@ -64,10 +109,29 @@ export default function SingleReviewPage() {
             )}
           </div>
-          <div className="Add Comment Field">Form will go here</div>
+          <form className="Add Comment Field">
```
I'd also consider extracting the form into its own component, although not as big as the other sections, just for readability too

Poonam-raj commented on src/Routed/SingleReviewPage.jsx:
```diff
@@ -64,10 +109,29 @@ export default function SingleReviewPage() {
+            </label>
+            {addedCommentText.trim().length !== 0 && (
+              <button onClick={handleAddCommentClick}>Add Comment</button>
+            )}
```
I like how the submit button only pops up when we type something in, really cool take on validation

But I'd really like some more obvious validation from an accessibility point of view as there's no communication about what is wrong with submitting when the form input is empty. Currently when I try to enter in the input box when it's empty (pressing my enter key) it reloads the page which almost feels like the form worked even when nothing is posted.I'd prefer to not have anything happen and a message to come up to tell me it's not possible

Poonam-raj commented on src/Routed/SingleReviewPage.jsx:
```diff
@@ -64,10 +109,29 @@ export default function SingleReviewPage() {
             {isLoading ? (
```
Line 106 -> the way we render the number of comments isn't dynamic e.g. when I post a new comment it doesn't change unless I refresh but I'd way prefer to see this number change when I post or delete straight away

Poonam-raj commented on src/Routed/AllReviewsPage.jsx:
```diff
@@ -15,11 +15,71 @@ export default function AllReviewsPage() {
+      }
+    );
+  };
+
```
I'd love some sort of visual feedback to tell me what we are currently sorting by

Poonam-raj reviewed (changes requested):
Few things I'd like you to look again here, please see my comments - thank you :D

ezisberta commented:
Hi Poonam, thanks for your feedback! I've extracted the comment section into its own component as you suggested, however, with the vote field, I don't feel like it takes too much space at the moment and there aren't further to dos that suggest it's going to grow, so I'll leave it as it is for now. 
About extracting the form, I'm finding it a bit tricky at the moment, and since its more of a organisation feature than a functionality one, I am going to look closer into it once I've completed the remaining tasks and added some CSS to the project :)
About the form validation, I've stopped it from refreshing the page when enter is pressed, And I've also added a message that briefly replaces the form once it's been submitted successfully.
